libraries/WiFi/src/WiFiGeneric.cpp

```diff
@@ -1051,12 +1063,13 @@ esp_err_t WiFiGenericClass::_eventCallback(arduino_event_t *event)
         }
     } else if(event->event_id == ARDUINO_EVENT_WIFI_STA_STOP) {
         WiFiSTAClass::_setStatus(WL_STOPPED);
-        clearStatusBits(STA_STARTED_BIT | STA_CONNECTED_BIT | STA_HAS_IP_BIT | STA_HAS_IP6_BIT);
+        clearStatusBits(STA_STARTED_BIT | STA_CONNECTED_BIT | STA_HAS_IP_BIT | STA_HAS_IP6_BIT | STA_HAS_IP6_GLOBAL_BIT);
     } else if(event->event_id == ARDUINO_EVENT_WIFI_STA_CONNECTED) {
+        if (getStatusBits() & STA_WANT_IP6_BIT){
+            esp_netif_create_ip6_linklocal(get_esp_interface_netif(ESP_IF_WIFI_STA));
```

Comment:
Putting back the problem where it belongs. Sorry for the confusion.

I'm seeing a race condition here. `esp_netif_create_ip6_linklocal` returns `-1` because:

`get_esp_interface_netif(ESP_IF_WIFI_AP)->lwip_netif->flags` has the flag `NETIF_FLAG_UP` not set when it is called. However if I add
`delay(1)`
the flag `NETIF_FLAG_UP` is set and the link-local address is successfully assigned.

I have no clue why and I have never seen this behavior with our previous code.

Any ideas?

Reply:
I did not check the return value, but other than that IPv6 works normally over STA and ETH

Reply:
I did update to the later platform builds @Jason2866  made including later IDF code and the final merged IPv6 code and ran into similar issues as described by @s-hadinger 
But now the really strange part(s)....

- ESP32-S2 is getting IPv6 just fine, no need to add an extra delay
- ESP32 (classic) is not getting IPv6. Neither on WiFi nor on Ethernet (LAN8720A) 

N.B. This was working fine about 2 weeks ago when I was running platform builds before your code was merged, but this also had older IDF code)

Now I added this delay to the WiFi connected event and it is showing IPv6 addresses just as before.
So I also added it to the Ethernet code and now the really funky part.
The ETH class doesn't show any IP address (`0.0.0.0` for IPv4 and `::` for IPv6) when I query it, but the ESP is accessible via Ethernet on its assigned DHCP IP.
Still have to look into this a bit more to see if it does also get IPv6.
If it does, then this is more like an event from IDF code is not received/processed by the Arduino code and not about not getting the IP assigned.

Also I will check the other ESP32-variants like C3/C2/C6/S3 to see if those work just fine without this delay.

Reply:
About the `0.0.0.0` / `::` issue for Ethernet....
That was a PEBKAC problem.
Anyway the delay is still needed for ESP32 classic boards.
Still have to check the other ESP32-variants to see if these also need this delay.